File: app/Models/Transaction.php
<?php

namespace App\Models;

use App\Enums\TransactionType;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Transaction extends Model
{
    use HasFactory;

    protected $fillable = [
        'type',
        'date',
        'amount',
        'description'
    ];

    public function casts()
    {
        return [
            'type' => TransactionType::class,
            'date' => 'date',
            'amount' => 'decimal:2',
        ];
    }

    public function getDateAttribute($value)
    {
        return $this->asDate($value)->format('Y-m-d');
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function category()
    {
        return $this->belongsTo(Category::class);
    }
}

File: Models/Entities/Transaction.cs
using System.ComponentModel.DataAnnotations.Schema;
using Florin_Back.Enums;

namespace Florin_Back.Models.Entities;

public class Transaction
{
    public long Id { get; set; }
    public long UserId { get; set; }
    public long CategoryId { get; set; }
    public TransactionType Type { get; set; }
    public DateOnly Date { get; set; }
    public decimal Amount { get; set; }
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    [ForeignKey(nameof(UserId))]
    public User? User { get; set; }

    [ForeignKey(nameof(CategoryId))]
    public Category? Category { get; set; }
}
